### PR TITLE
fix: [BUG] - Commands not loaded in dev #853

### LINKED ISSUE #853: [BUG] - Commands not loaded in development
### What bug did you encounter?

When refactoring Becca for production, we modified the logic such that she would not reload her commands on startup.

This is ideal for production, but not ideal for development.

### What did you expect to happen?

In the `index.ts`, we will need to add a step after loading the commands to check if `NODE_ENV !== "production"`, and if true, call the `registerCommands` utility. 

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gitaalekhyapaul**
# Pull Request

## Description
Added command registration in development mode

## Related Issue

Closes #853 

## Documentation

For _any_ version updates, please verify if the [documentation page](https://www.beccalyria.com/discord-documentation) needs an update. If it does, please [create an issue there](https://github.com/BeccaLyria/discord-documentation/issues/new?assignees=nhcarrigan&labels=%F0%9F%9A%A6+status%3A+awaiting+triage&template=update.md&title=%5BUPDATE%5D) to ensure it is not forgotten.

- [x] My contribution does NOT require a documentation update.
- [ ] My contribution DOES require a documentation update.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -63,6 +64,11 @@ const initialiseBecca = async () => {
     return;
   }
 
+  if (process.env.NODE_ENV !== "production") {
+    beccaLogHandler.log("debug", "Registering commands in development...");
+    await registerCommands(Becca);
+  }
+
```
Comment (nhcarrigan) [inline src/index.ts:71]:
```suggestion
  if (process.env.NODE_ENV !== "production") {
    beccaLogHandler.log("debug", "Registering commands in development...");
    const success = await registerCommands(Becca);
    if (!success) {
      beccaLogHandler.log("error", "failed to register commands.");
      return;
    }
  }

```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
I did not mention this in the issue (I should have), but we can assert that the command registration succeeds.

File: src/index.ts (hunk 1 above)
Comment (gitaalekhyapaul) [reply src/index.ts]: Added in https://github.com/BeccaLyria/discord-bot/pull/859/commits/6916a74c446ee8e103c102e0598f66ec65e82984

**Review by nhcarrigan — APPROVED** [review]
Looks good to me!

**Comment by nhcarrigan** [comment]
## Congratulations

Congratulations on making your first pull request to our project. We have reviewed your contributions and are happy to accept them.

We hope you enjoyed the contribution process, and encourage you to watch for issues labelled `help wanted` for additional opportunities to contribute.

**Pull request merged**